workloads/pytorch/recommendation/recoder/model.py

```diff
@@ -456,7 +480,10 @@ def _train(self, train_dataloader, val_dataloader,
 
       if model_checkpoint_prefix and \
           ((checkpoint_freq > 0 and epoch % checkpoint_freq == 0) or epoch == num_epochs):
-        self.save_state(model_checkpoint_prefix)
+        if self._enable_gavel_iterator:
+          self.save_state(model_checkpoint_prefix, train_dataloader)
```

Comment:
Indentation in this file looks weird: 2 spaces in some places, 4 spaces in others